spring-boot-starter/starter-security/src/main/java/org/camunda/bpm/spring/boot/starter/security/oauth2/CamundaSpringSecurityOAuth2AutoConfiguration.java

```diff
@@ -0,0 +1,94 @@
+import org.springframework.boot.autoconfigure.security.oauth2.client.ClientsConfiguredCondition;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.core.Ordered;
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.web.SecurityFilterChain;
+
+import java.util.Map;
+
+@AutoConfigureOrder(CamundaSpringSecurityOAuth2AutoConfiguration.CAMUNDA_OAUTH2_ORDER)
+@AutoConfigureAfter({ CamundaBpmAutoConfiguration.class, SpringProcessEngineServicesConfiguration.class })
+@ConditionalOnBean(CamundaBpmProperties.class)
+@Conditional(ClientsConfiguredCondition.class)
+public class CamundaSpringSecurityOAuth2AutoConfiguration {
+
+  private static final Logger logger = LoggerFactory.getLogger(CamundaSpringSecurityOAuth2AutoConfiguration.class);
+  public static final int CAMUNDA_OAUTH2_ORDER = Ordered.HIGHEST_PRECEDENCE + 100;
+  private final String webappPath;
+
+  public CamundaSpringSecurityOAuth2AutoConfiguration(CamundaBpmProperties properties) {
+    WebappProperty webapp = properties.getWebapp();
+    this.webappPath = webapp.getApplicationPath();
+  }
+
+  @Bean
+  public FilterRegistrationBean<?> webappAuthenticationFilter() {
+    FilterRegistrationBean<Filter> filterRegistration = new FilterRegistrationBean<>();
+    filterRegistration.setName("Container Based Authentication Filter");
+    filterRegistration.setFilter(new ContainerBasedAuthenticationFilter());
+    filterRegistration.setInitParameters(Map.of(
+        ProcessEngineAuthenticationFilter.AUTHENTICATION_PROVIDER_PARAM, OAuth2AuthenticationProvider.class.getName()));
+    // make sure the filter is registered after the Spring Security Filter Chain
+    filterRegistration.setOrder(SecurityProperties.DEFAULT_FILTER_ORDER + 1);
+    filterRegistration.addUrlPatterns(webappPath + "/app/*");
+    filterRegistration.setDispatcherTypes(DispatcherType.REQUEST);
+    return filterRegistration;
+  }
+  
+  @Bean
+  public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+    logger.info("Enabling Camunda Spring Security oauth2 integration");
+    
+    http.authorizeHttpRequests(c -> c
+            .requestMatchers(webappPath + "/app/**").authenticated()
+            .requestMatchers(webappPath + "/api/**").authenticated()
+            .anyRequest().permitAll()
+        )
+        .oauth2Login(Customizer.withDefaults())
+        .oauth2Login(Customizer.withDefaults())
+        .oidcLogout(Customizer.withDefaults())
+        .oauth2Client(Customizer.withDefaults())
+        .cors(AbstractHttpConfigurer::disable)
+        .csrf(AbstractHttpConfigurer::disable);
```

Comment:
❓ Will this be configurable in the next tickets?

Reply:
As discussed in our call, for csrf we stick with disable. For CORS, we can check if we can and want to make it configurable.

Reply:
I checked the cors and it looks like it doesn't matter if it's disabled, withDefaults or not even specified. So I'll drop that line, too.
If customers need a CORS filter in Boot they can add it easily and in Run we have already support.